src/Datadog.Trace/ExtensionMethods/SpanExtensions.cs

```diff
@@ -67,6 +71,26 @@ public static void AddTagsFromDbCommand(this Span span, IDbCommand command)
             }
         }
 
+        internal static void ApplyHeaderTags(this Span span, IHeadersCollection headers, IDictionary<string, string> headerTags)
```

Comment:
We usually talk about "setting" tags, not "applying" them. Consider renaming to `SetHeaderTags()`. Level of care low.